File: src/models/request.js
const mongoose = require('mongoose');

const requestSchema = new mongoose.Schema({
  method: String,
  url: String,
  headers: Object,
  body: Object,
  response: Object,
  status: { type: String, enum: ['pending', 'completed', 'failed'], default: 'pending' },
  statusCode: Number,
  error: String,
  timestamp: { type: Date, default: Date.now },
  completedAt: Date
});

module.exports = mongoose.model('Request', requestSchema); 

File: python-api-testing/app/models/request.py
from datetime import datetime
from typing import Optional, Dict, Any
from pydantic import BaseModel, Field
from bson import ObjectId

class PyObjectId(ObjectId):
    @classmethod
    def __get_validators__(cls):
        yield cls.validate

    @classmethod
    def validate(cls, v):
        if not ObjectId.is_valid(v):
            raise ValueError("Invalid ObjectId")
        return ObjectId(v)

class RequestBase(BaseModel):
    method: str
    url: str
    headers: Dict[str, Any]
    body: Optional[Dict[str, Any]] = None
    status: str = "pending"
    statusCode: Optional[int] = None
    response: Optional[Dict[str, Any]] = None
    error: Optional[str] = None
    timestamp: datetime = Field(default_factory=datetime.utcnow)
    completedAt: Optional[datetime] = None

class RequestInDB(RequestBase):
    id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")

    class Config:
        allow_population_by_field_name = True
        json_encoders = {
            ObjectId: str,
            datetime: lambda dt: dt.isoformat()
        } 